controllers/managedosversionchannel_controller.go

```diff
@@ -187,12 +187,12 @@ func (r *ManagedOSVersionChannelReconciler) reconcile(ctx context.Context, manag
 
 	if readyCondition.Status == metav1.ConditionTrue {
 		logger.Info("synchronization already done", "lastSync", lastSync)
-		return ctrl.Result{}, nil
+		return ctrl.Result{RequeueAfter: time.Until(lastSync.Add(interval))}, nil
```

Comment:
IMHO this shouldn't be needed, but it certainly does not hurt and helps on making the logic more robust. The unit test verifying the automatic resync after the interval passes without this.

Reply:
Yeah, I was wondering if this could let us to queue extra, unneeded reconcile loops, but seems in practice never happens. Moreover, if we ever would reconcile once more, nothing bad could happen 👍🏼